src/helper/Reply.cc

```diff
@@ -192,6 +192,20 @@ Helper::Reply::emptyBuf() const
     return empty;
 }
 
+void
+Helper::checkForUnsupportedAnnotations(const NotePairs &notes, const char *context)
+{
+    const auto entries = notes.expandListEntries(nullptr);
+    for (const auto &e: entries) {
```

Comment:
Please ignore this premature change request until the caller design becomes stable. This function interface and implementation may change quite a bit by then...

----

This copies all expanded entries, which feels excessive. It is also strange that we are asking `notes` to expand value lists when we do not actually care about annotation values at all. Why not iterate `notes` themselves?! We already have Notes::begin()/end() and can add constant versions of those iterators for this range-based for loop.